server/catgenome/src/test/java/com/epam/catgenome/manager/reference/MotifSearchManagerTest.java

```diff
@@ -0,0 +1,680 @@
+                .chromosomeId(hpGenomeTestReference.getChromosomes().get(1).getId())
+                .motif(EXTENDED_TEST_GENOME_MOTIF)
+                .searchType(MotifSearchType.WHOLE_GENOME)
+                .pageSize(PAGE_SIZE)
+                .strand(StrandSerializable.POSITIVE)
+                .build();
+        MotifSearchResult search = motifSearchManager.search(att);
+        Set<String> chromosomes = search.getResult().stream().map(Motif::getContig)
+                .collect(Collectors.toSet());
+        Assert.assertTrue(chromosomes.size() > 1);
+    }
+
+    @Test
+    @Transactional(propagation = Propagation.REQUIRES_NEW, rollbackFor = Exception.class)
+    public void checkThatPositionIsNullWhenWeSearchByExtMotifForLargeNumberOfResultsTest() {
+        MotifSearchRequest att = MotifSearchRequest.builder()
+                .referenceId(hpGenomeTestReference.getId())
+                .chromosomeId(hpGenomeTestReference.getChromosomes().get(1).getId())
+                .startPosition(1)
+                .motif(EXTENDED_TEST_GENOME_MOTIF)
+                .searchType(MotifSearchType.WHOLE_GENOME)
+                .pageSize(BIG_PAGE_SIZE)
+                .strand(StrandSerializable.POSITIVE)
+                .build();
+        MotifSearchResult search = motifSearchManager.search(att);
+        Assert.assertNull(search.getPosition());
+        Assert.assertFalse(search.getResult().isEmpty());
+    }
+
+    @Test
+    @Transactional(propagation = Propagation.REQUIRES_NEW, rollbackFor = Exception.class)
+    public void checkThatPositionIsNullWhenWeSearchFullGenomeByExtMotifButDataOnlyInFirstChrTest() {
+        MotifSearchRequest att = MotifSearchRequest.builder()
+                .referenceId(testWgTestReference.getId())
+                .startPosition(1)
+                .chromosomeId(testWgTestReference.getChromosomes().get(0).getId())
+                .motif("CC+")
+                .searchType(MotifSearchType.WHOLE_GENOME)
+                .pageSize(PAGE_SIZE)
+                .strand(StrandSerializable.POSITIVE)
+                .build();
+        MotifSearchResult search = motifSearchManager.search(att);
+        Assert.assertNull(search.getPosition());
+        Assert.assertFalse(search.getResult().isEmpty());
+    }
+
+    @Test(expected = Test.None.class)
+    @Transactional(propagation = Propagation.REQUIRES_NEW, rollbackFor = Exception.class)
+    public void shouldNotFailWithOutOfMemoryErrorWhenWeSearchShortestMotif() {
+        MotifSearchRequest att = MotifSearchRequest.builder()
+                .referenceId(testReference.getId())
+                .startPosition(1)
+                .chromosomeId(testChromosome.getId())
+                .motif("[acgtn]")
+                .searchType(MotifSearchType.WHOLE_GENOME)
+                .pageSize(PAGE_SIZE)
+                .strand(null)
+                .build();
+        motifSearchManager.search(att);
+    }
+
+    @Test(expected = IllegalStateException.class)
```

Comment:
It should be IllegalArgumentException instead of IllegalStateException at 666 line

Reply:
+